src/generic/LaurentSeries.jl

```diff
@@ -400,6 +400,10 @@ function renormalize!(z::LaurentSeriesElem)
    return nothing
 end
 
+function characteristic(a::LaurentSeriesRing{T}) where T <: RingElement
+   return characteristic(base_ring(a))
+end
```

Comment:
Similar suggestion (it's lighter on the eyes): `characteristic(a::LaurentSeriesRing) = characteristic(base_ring(a))`

Reply:
It overflows the 80 character line.

Reply:
I though the limit was 92, but my suggestion here takes 68 characters.

Reply:
Only if you remove the "where", which I see no overwhelming reason to do. It is a distraction.

Reply:
> It is a distraction.

Seriously, I will sutt up after this one, but the `where` **is** the distraction.

Reply:
I was going home at 5pm. It's nearly 6pm now. It's wasted heaps of my time arguing about something that will have absolutely zero impact on any open tickets! We have a computer algebra system to write. We can't spend all our time amending code according to conventions that are the opposite to the ones we already use!

Reply:
I'm sorry you take it like this. I regret my last comment above, which was not constructive. Otherwise, I'm sincerely sorry that you feel you wasted your time, but I won't take any blame for this. It's your responsibility if you engage in a discussion which you consider useless.

My motivation in reviewing a PR is to make this project as successful as possible within my understanding of the constraints. My goal is _not_ to make cosmetic changes just according to my current personal preferences. If a review comment of mine is dismissed and answered with something like "no, we just do it this way", I won't argue, I understand that my decisional power is null, and I'm no more happy than you are to waste my time in pointless argumentations. On the other hand, if my comment is met with arguments to explain why this is the way it is, it's only natural to expect counter-arguments if anyone has to share any.

I find your comment above very negatively dismissive, it suggests that my contributions are not welcomed here, at least unless I'm sure they have sufficient value in the eyes of the maintainers. I learn slowly what that means, but again, I usually contribute what I think is valuable to the project.

> We can't spend all our time amending code according to conventions that are the opposite to the ones we already use!

 In the present case of this PR, I initially didn't get from the comments that it was a convention, it really looked like an oversight. So sure, I agree that "We can't spend all our time amending code according to conventions that are the opposite to the ones we already use", but I didn't suggest such a thing here. And I'm already aware that you don't like suggestions to change these custom conventions, so I usually try to refrain from from that.

Reply:
@rfourquet Your contributions and suggestions are always welcome here. You surely know that by now. In fact, I didn't initially try to just shut down your suggestion precisely because I wanted to hear you out on your reasons. But in my opinion it is quite obvious that in our code we do this all over the place. I am sure the reason is not obvious, and it is done inconsistently.

The history of this is that we initially started doing what you suggested, because we understood that it was making lighter work for the type checker and that there should be some performance benefit. It became clearer over the time (to me) that this was likely not the case (I did a lot of experiments in this direction in the MPoly module). And over time we have drifted much more towards putting this information in for mathematical correctness and documentation.

It could change again in the future as people see a different dimension to the project. But as always, we would need a concensus on the matter from everyone (including you). At the current point in time, I personally don't agree to the change. If I am overruled, and we have the resources to make such a change globally across the package, I will accept the decision. But as I have suggested over and over again, I think that we have much, much more important things to worry about right now. We really have to start focusing our developer effort on things that will maximise the benefit for our users and the grant that pays our salaries. We need to prioritise things on this basis, not on how beautiful things will be if we make this or that change.

Reply:
> it is quite obvious that in our code we do this all over the place 

Indeed, but it looks like the result of a misunderstanding. I've never seen this style elsewhere and it seems so heavyweigthed and counter-productive to me that I could not imagine it was a deliberate convention. Like all the docstrings prefixed with `@doc Markdown.doc`: I clearly thought it's a result of the belief that it's necessary (it's probably only useful for when there is latex and backslashes, and then I think the documented way is to use a `raw` string, but anyway). So of course I wonder now whether this could be another implicit convention. 

> But as I have suggested over and over again, I think that we have much, much more important things to worry about right now. [...]

Yes, and I wonder why you feel the need to patronize me on this again. I've made clear that I didn't suggest anything here that would cost developer time. I only suggested to amend _this_ PR (not all the repository), because, in my opinion, it saves maintenance effort in the future, and it makes it easier to read the code. So my goal is the opposite of wasting developer time, and this is not about beauty. The idea that I would want us to waste time on useless stuff is imagined. For example, I suggested [above](https://github.com/Nemocas/AbstractAlgebra.jl/pull/544#discussion_r353287412) that "It should be also removed from the docstrings", because I feel it's a documentation issue (and actually I had started preparing a doc PR for something quite related some weeks ago). But I didn't ask anyone to do it, even less to do it all at once, and it was before I understood it was a delibarate convention.

As a kind of fun side note, our first interaction seems to have been on more or less this topic, 5 years ago! 
https://groups.google.com/d/msg/sage-devel/1gPkeL_X5dw/rKRglDI8n1gJ

Reply:
I remember that discussion, though I hadn't remembered it was you that replied to me there. I did remember Robert being so disappointed at my hack!

Anyway, clearly we disagree on some things not germane to this PR and we should at least discuss them elsewhere (IRL). I hope that you don't really feel patronised. I will always interpret a suggestion to follow a convention as an invitation to make the change everywhere. Otherwise it isn't a convention and I won't enforce it.

In any case, my bad. I should not have used the word convention here. We do not have any convention on this issue. What I should have said is, "it has been a common practice". So I apologise if I made more out of this than should have been.

Having said that, this is not a change we support, so I won't make the change here either. I appreciate that I misunderstood your motivations here, and I do appreciate you making suggestions. Please accept my apology for making more of this than I should have.

Reply:
By the way, is there anything else remaining which is germane to the PR that I haven't sorted yet. It would be nice to be able to merge the PR.

Reply:
Thanks for your comprehension. I must admit that I still don't fully understand the subtlety with a "common practice" which is not really a convention but which must be consistent. But I totally agree we should discuss these not-germane-to-the-PR things directly IRL, if/when needed.

Reply:
You understand correctly. A "convention" should be enforced. A "common practice" has not been and will not be unless we make a decision to make it a convention. I should not have used the word convention.